### PR TITLE
Proper handling of local onload callback

### PR DESCRIPTION
**Pull request by krojew**

### CODE REVIEW

**Comment by krojew** [comment]
ticket #844

**Comment by obiot** [comment]
:+1 :)

File: src/loader/loader.js [hunk 1]
```diff
@@ -35,17 +35,17 @@
...
                 // wait 1/2s and execute callback (cheap workaround to ensure everything is loaded)
-                if (api.onload) {
+                if (onload || api.onload) {
                     // make sure we clear the timer
                     clearTimeout(timerId);
                     // trigger the onload callback
-
-                    var onload = api.onload;
+                    // we call either the supplied callback (which takes precedence) or the global one
+                    var callback = onload || api.onload;
```
Comment (parasyte) [inline src/loader/loader.js:46]: This can be hoisted to the top of function scope, so you don't have to falsy-check `onload` twice.

Comment (obiot) [reply src/loader/loader.js:46]: i would even say that since checkLoadStatus is always called passing onload, there is no real need for that

Comment (parasyte) [reply src/loader/loader.js:46]: Excellent observation! The function is private (undocumented), so none of the fallback logic is even necessary.

Comment (obiot) [reply src/loader/loader.js:46]: thanks, I still have my moments sometimes :P:P:P

**Pull request merged**